Tasks/QuickPerfTest/CltTasksUtility.ps1

```diff
@@ -0,0 +1,191 @@
+function InvokeRestMethod($headers, $contentType, $uri , $method= "Get", $body)
+{
+  $ServicePoint = [System.Net.ServicePointManager]::FindServicePoint($uri)
+  $result = Invoke-RestMethod -ContentType "application/json" -UserAgent $global:userAgent -TimeoutSec $global:RestTimeout -Uri $uri -Method $method -Headers $headers -Body $body
+  $ServicePoint.CloseConnectionGroup("")
+  return $result
+}
+
+function ComposeTestDropJson($name, $duration, $homepage, $vu, $geoLocation)
+{
+$tdjson = @"
+{
+    "dropType": "InplaceDrop",
+    "loadTestDefinition":{
+        "loadTestName":"$name",
+        "runDuration":$duration,
+        "urls":["$homepage"],
+        "browserMixs":[
+            {"browserName":"Internet Explorer 11.0","browserPercentage":60.0},
+            {"browserName":"Chrome 2","browserPercentage":40.0}
+        ],
+        "loadPatternName":"Constant",
+        "maxVusers":$vu,
+        "loadGenerationGeoLocations":[
+            {"Location":"$geoLocation","Percentage":100}
+        ]
+    }
+}
+"@
+
+    return $tdjson
+}
+
+function CreateTestDrop($headers, $dropJson, $CltAccountUrl)
+{
+    $uri = [String]::Format("{0}/_apis/clt/testdrops?api-version=1.0", $CltAccountUrl)
+    $drop = InvokeRestMethod -contentType "application/json" -uri $uri -method Post -headers $headers -body $dropJson
+
+    return $drop
+}
+
+function GetTestDrop($headers, $drop, $CltAccountUrl)
+{
+    $uri = [String]::Format("{0}/_apis/clt/testdrops/{1}?api-version=1.0", $CltAccountUrl, $drop.id)
+    $testdrop = InvokeRestMethod -contentType "application/json" -uri $uri -headers $headers
+
+    return $testdrop
+}
+
+function UploadTestDrop($testdrop)
+{
+    $uri = New-Object System.Uri($testdrop.accessData.dropContainerUrl)
+    $sas = New-Object Microsoft.WindowsAzure.Storage.Auth.StorageCredentials($testdrop.accessData.sasKey)
+    $container = New-Object Microsoft.WindowsAzure.Storage.Blob.CloudBlobContainer($uri, $sas)
+
+    return $container
+}
+
+function GetTestRuns($headers, $CltAccountUrl)
+{
+    $uri = [String]::Format("{0}/_apis/clt/testruns?api-version=1.0", $CltAccountUrl)
+    $runs = InvokeRestMethod -contentType "application/json" -uri $uri -headers $headers
+
+    return $runs
+}
+
+function GetTestRunUri($testRunId, $headers, $CltAccountUrl)
+{
+ $uri = [String]::Format("{0}/_apis/clt/testruns/{1}?api-version=1.0", $CltAccountUrl,$testRunId)
+ $run = InvokeRestMethod -contentType "application/json" -uri $uri -headers $headers
+ 
+ return $run.WebResultUrl
+}
+
+function RunInProgress($run)
+{
+    return $run.state -eq "queued" -or $run.state -eq "inProgress"
+}
+
+function MonitorTestRun($headers, $run, $CltAccountUrl)
+{
+    $uri = [String]::Format("{0}/_apis/clt/testruns/{1}?api-version=1.0", $CltAccountUrl, $run.id)
+    $prevState = $run.state
+    $prevSubState = $run.subState
+    Write-Output ("Load test '{0}' is in state '{1}|{2}'." -f  $run.name, $run.state, $run.subState)
+
+    do
+    {
+        Start-Sleep -s 5
```

Comment:
> Start-Sleep -s 5 [](start = 7, length = 17)

increase the polling duration.. 5 seconds is too low.. it could be the sampling rate till the run reaches inprogress and can be higher later..